apps/explorer/lib/explorer/third_party_integrations/universal_proxy.ex

```diff
@@ -0,0 +1,314 @@
+
+      _ ->
+        {"Unexpected error when calling proxied endpoint", 500}
+    end
+  end
+
+  @doc """
+  Parses the proxy parameters and constructs an API request.
+
+  This function takes a map of proxy parameters and uses the configuration
+  to determine the platform-specific settings, such as the base URL, API key,
+  HTTP method, and endpoint parameters. It returns a map representing the API
+  request.
+
+  ## Parameters
+
+    - `proxy_params` (map): A map containing the proxy parameters, including
+      the `platform_id` which is used to identify the platform configuration.
+
+  ## Returns
+
+    - A map representing the API request with the following keys:
+      - `:url` (String.t | nil): The constructed URL for the API request.
+      - `:body` (String.t): The body of the API request (currently an empty string).
+      - `:headers` (list): A list of headers for the API request (currently empty).
+      - `:method` (atom): The HTTP method for the API request (e.g., `:get`, `:post`).
+      - `:platform_config` (map | nil): The configuration for the specified platform.
+      - `:platform_id` (String.t | nil): The ID of the platform.
+
+  ## Notes
+
+    - If the platform configuration or required fields are missing, the `:url`
+      in the returned map will be `nil`.
+    - The function delegates further processing of the API key and endpoint
+      parameters to `parse_endpoint_api_key/3` and `parse_endpoint_params/3`.
+  """
+  @spec parse_proxy_params(map()) :: api_request()
+  def parse_proxy_params(proxy_params) do
+    config = config()
+    platform_id = proxy_params["platform_id"]
+
+    platform_config = config["platforms"] && config["platforms"][platform_id]
+    endpoint_params = (platform_config && platform_config["endpoints"]["base"]["params"]) || []
+    endpoint_api_key = platform_config && platform_config["api_key"]
+
+    method = parse_method(platform_config)
+
+    raw_url =
+      with true <- not is_nil(platform_config),
+           true <- not is_nil(platform_config["base_url"]),
+           endpoints = platform_config["endpoints"],
+           true <- not is_nil(endpoints),
+           base_endpoint = endpoints["base"],
+           true <- not is_nil(base_endpoint),
+           true <- not is_nil(base_endpoint["path"]) do
+        platform_config["base_url"] <> base_endpoint["path"]
+      else
+        _ ->
+          nil
+      end
+
+    %{
+      url: raw_url,
+      body: "",
+      headers: [],
+      method: method,
+      platform_config: platform_config,
+      platform_id: platform_id
+    }
+    |> parse_endpoint_api_key(endpoint_api_key, platform_id)
+    |> parse_endpoint_params(endpoint_params, proxy_params)
+  end
+
+  @spec parse_method(map() | nil) :: atom() | nil
+  defp parse_method(nil), do: nil
+
+  # sobelow_skip ["DOS.StringToAtom"]
+  defp parse_method(platform_config) do
+    raw_method = String.to_atom(String.downcase(platform_config["endpoints"]["base"]["method"]))
```

Comment:
nitpick: Can we use `to_existing_atom/0` here?

```suggestion
  defp parse_method(platform_config) do
    raw_method = 
      platform_config["endpoints"]["base"]["method"]
      |> String.downcase()
      |> String.to_existing_atom()
```

Reply:
The reason I didn't use `String.to_existing_atom` is this:
```
iex(2)> String.to_existing_atom("patch")
** (ArgumentError) errors were found at the given arguments:

  * 1st argument: not an already existing atom

    :erlang.binary_to_existing_atom("patch", :utf8)
    iex:2: (file)
```

Reply:
oh, sorry, I see.